extensions/libarchive/MergeContent.cpp

```diff
@@ -342,6 +362,56 @@ std::shared_ptr<core::FlowFile> ZipMerge::merge(core::ProcessContext *context, c
   return flowFile;
 }
 
+void AttributeMerger::mergeAttributes() {
+  std::map<std::string, std::string> commonAttributes = getCommonAttributes();
+
+  for (const auto& pair : commonAttributes) {
+    session_->putAttribute(mergeFlow_, pair.first, pair.second);
+  }
+}
+
+std::map<std::string, std::string> AttributeMerger::getCommonAttributes() {
+  std::map<std::string, std::string> commonAttributes;
```

Comment:
This could potentially simplify things:
https://en.cppreference.com/w/cpp/algorithm/set_intersection

Reply:
This would require creating a new map for every intersection's output as it seems I cannot update the same map for multiple intersections, that would be costly.

Reply:
Not necessarily, if you move into the algo:
https://en.cppreference.com/w/cpp/iterator/make_move_iterator

Reply:
Are you suggesting something like this?

`auto attr = flow->getAttributes();`
`std::map<std::string, std::string> tmpMerged;`
`std::set_intersection(std::make_move_iterator(mergedAttributes.begin()), std::make_move_iterator(mergedAttributes.end()), std::make_move_iterator(attr.begin()), std::make_move_iterator(attr.end()), std::inserter(tmpMerged, tmpMerged.begin()));`
`mergedAttributes = std::move(tmpMerged);`

Wouldn't it invalidate the original attributes of the original flowfile or it doesn't matter in this case?

Reply:
Oh now I see we can consume the flow attributes as it's just a copy never mind then.

Reply:
Probably the not the most efficient, but very readable:

Assuming 
 1. A := The current common attributes map
 1. B := The current flow-file attribute map
 1. X := Keys(A) ∩ Keys(B)

Then the two kinds of mergers you want is:
 1. **Keep All Unique Attributes:** (Elements of A at X) ∩ (Elements of B at X)
 1. **Keep Only Common Attributes:** A ∩ B

Reply:
The only problem in the unique attribute case is after the result of the first intersection we remove the attributes with conflicting values, then in the next processing of the next flowfile attributes the removed attribute can reappear in that flowfile and we handle it as a unique attribute adding it to the final merged attribute map. (More simply: missing the removed attributes handling)

Reply:
I used the set_intersection for the common attributes, but kept the original solution for the unique attributes